### PR TITLE
[PHI Decoupling]Remove memory header (Part2)

### PR DESCRIPTION
**Pull request by YuanRisheng**
### PR types

Others
### PR changes

Others
### Describe

memory头文件解耦第二个PR，移除：paddle/fluid/memory/memory.h 和 paddle/fluid/memory/memcpy.h
memory解耦的第一个PR详见：https://github.com/PaddlePaddle/Paddle/pull/50419

### CODE REVIEW

File: paddle/phi/kernels/gpu/randint_kernel.cu [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include "paddle/phi/kernels/funcs/distribution_helper.h"
 
 // See Note [ Why still include the fluid headers? ]
```
Comment (jiahy0825) [inline paddle/phi/kernels/gpu/randint_kernel.cu:23]: The comment here can also be deleted

File: paddle/phi/kernels/xpu/full_kernel.cc [hunk 2]
```diff
@@ -24,7 +24,7 @@
 #include "paddle/phi/core/visit_type.h"
 
 // See Note [ Why still include the fluid headers? ]
```
Comment (jiahy0825) [inline paddle/phi/kernels/xpu/full_kernel.cc:26]: Same as above

**Review by jiahy0825 — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

File: paddle/phi/kernels/gpu/randint_kernel.cu (hunk 1 above)
Comment (YuanRisheng) [reply paddle/phi/kernels/gpu/randint_kernel.cu:23]: 谢谢，下个pr修改

File: paddle/phi/kernels/xpu/full_kernel.cc (hunk 2 above)
Comment (YuanRisheng) [reply paddle/phi/kernels/xpu/full_kernel.cc:26]: 谢谢，下个pr修改

**Review by ZzSean — APPROVED** [review]
LGTM for CI-OP-Benchmark

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**